File: lib_common/src/main/java/com/yyxnb/common/interfaces/IData.java
package com.yyxnb.common.interfaces;

/**
 * 接口返回封装类
 *
 * @param <T>
 */
public interface IData<T> {

    default int id() {
        return hashCode();
    }

    String getCode();

    String getMsg();

    T getResult();

    boolean isSuccess();
}

File: lib_common/src/main/java/com/yyxnb/common/interfaces/IData.kt
package com.yyxnb.common.interfaces

/**
 * 接口返回封装类
 *
 * @param <T>
 */
interface IData<T> {

    fun id(): Int {
        return hashCode()
    }

    val code: String?
    val msg: String?
    val result: T?
    val isSuccess: Boolean
}